src/model/operation/transform.js

```diff
@@ -1,644 +1,1895 @@
-/**
- * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
- * For licensing, see LICENSE.md.
- */
+import InsertOperation from './insertoperation';
+import AttributeOperation from './attributeoperation';
+import RenameOperation from './renameoperation';
+import MarkerOperation from './markeroperation';
+import MoveOperation from './moveoperation';
+import RemoveOperation from './removeoperation';
+import ReinsertOperation from './reinsertoperation';
+import RootAttributeOperation from './rootattributeoperation';
+import MergeOperation from './mergeoperation';
+import SplitOperation from './splitoperation';
+import WrapOperation from './wrapoperation';
+import UnwrapOperation from './unwrapoperation';
+import NoOperation from './nooperation';
+import Range from '../range';
+import Position from '../position';
+import compareArrays from '@ckeditor/ckeditor5-utils/src/comparearrays';
+
+const transformations = new Map();
+
+function setTransformation( OperationA, OperationB, transformationFunction ) {
+	let aGroup = transformations.get( OperationA );
+
+	if ( !aGroup ) {
+		aGroup = new Map();
+		transformations.set( OperationA, aGroup );
+	}
+
+	aGroup.set( OperationB, transformationFunction );
+}
+
+function getTransformation( a, b ) {
+	const OperationA = a.constructor;
+	const OperationB = b.constructor;
+
+	const aGroups = new Set();
+
+	let aGroup = transformations.get( OperationA );
+
+	if ( aGroup ) {
+		aGroups.add( aGroup );
+	}
+
+	for ( const Operation of transformations.keys() ) {
+		if ( a instanceof Operation ) {
+			aGroups.add( transformations.get( Operation ) );
+		}
+	}
+
+	for ( const group of aGroups ) {
+		if ( group.has( OperationB ) ) {
+			return group.get( OperationB );
+		}
+
+		for ( const Operation of group.keys() ) {
+			if ( b instanceof Operation ) {
+				return group.get( Operation );
+			}
+		}
+	}
+
+	return noUpdateTransformation;
+}
+
+function noUpdateTransformation( a ) {
+	return [ a.clone() ];
+}
+
+function updateBaseVersions( operations, baseVersion ) {
+	for ( const operation of operations ) {
+		operation.baseVersion = baseVersion++;
+	}
+
+	return operations;
+}
+
+function transform( a, b, context = { isStrong: false } ) {
```

Comment:
It is not strong tranform, but A is isStrong, the parameter name is missleading.

Reply:
The parameter name is not precise. It says which operation - a or b - is strong. Current name might suggest that the transformation itself is strong somehow.

Reply:
👍